src/commands/deploy/runPreDeployTask.ts

```diff
@@ -17,7 +17,7 @@ export async function runPreDeployTask(context: IActionContext, deployFsPath: st
     const preDeployTask: string | undefined = getWorkspaceSetting(preDeployTaskSetting, deployFsPath);
     if (preDeployTask && preDeployTask.startsWith('func:')) {
         const message: string = localize('installFuncTools', 'You must have the Azure Functions Core Tools installed to run preDeployTask "{0}".', preDeployTask);
-        if (!await validateFuncCoreToolsInstalled(message)) {
+        if (!await validateFuncCoreToolsInstalled(message, deployFsPath)) {
```

Comment:
Would the workspace setting be in the deployFsPath or the workspace folder root?  I thought for functions projects, most of the settings would be in the root folder.

Reply:
You don't have to pass the exact workspace folder path when getting a setting. VS Code will automatically drill down to the folder containing that path

Reply:
Also to be clear - VS Code ignores "settings.json" files that are not at the root anyways